WordPress/Classes/Services/BloggingPrompts/BloggingPromptsService.swift

```diff
@@ -260,6 +271,68 @@ private extension BloggingPromptsService {
         return Self.utcDateFormatter.date(from: dateString)
     }
 
+    // MARK: Prompts
+
+    /// Fetches a number of blogging prompts for the specified site from the v3 endpoint.
+    ///
+    /// - Parameters:
+    ///   - number: The number of prompts to query. When not specified, this will default to remote implementation.
+    ///   - fromDate: When specified, this will fetch prompts from the given date. When not specified, this will default to remote implementation.
+    ///   - ignoresYear: When set to true, this will convert the date to a custom format that ignores the year part. Defaults to true.
+    ///   - forceYear: Forces the year value on the prompt's date to the specified value. Defaults to the current year.
+    ///   - completion: A closure that will be called when the fetch request completes.
+    func fetchRemotePrompts(number: Int? = nil,
+                            fromDate: Date? = nil,
+                            ignoresYear: Bool = true,
+                            forceYear: Int? = nil,
+                            completion: @escaping (Result<[BloggingPromptRemoteObject], Error>) -> Void) {
+        let path = "wpcom/v3/sites/\(siteID)/blogging-prompts"
+        let requestParameter: [String: AnyHashable] = {
+            var params = [String: AnyHashable]()
+
+            if let number, number > 0 {
+                params["per_page"] = number
+            }
+
+            if let fromDate {
+                // convert to yyyy-MM-dd format in UTC timezone so users would get the same prompts everywhere.
+                var dateString = Self.utcDateFormatter.string(from: fromDate)
+
+                // when the year needs to be ignored, we'll transform the dateString to match the "--mm-dd" format.
+                if ignoresYear, !dateString.isEmpty {
+                    dateString = "-" + dateString.dropFirst(4)
+                }
+
+                params["after"] = dateString
+            }
+
+            if let forceYear = forceYear ?? fromDate?.dateAndTimeComponents().year {
+                params["force_year"] = forceYear
+            }
+
+            return params
+        }()
+
+        api.GET(path, parameters: requestParameter as [String: AnyObject]) { result, _ in
+            switch result {
+            case .success(let responseObject):
+                do {
+                    let data = try JSONSerialization.data(withJSONObject: responseObject, options: [])
+                    let decoder = JSONDecoder()
+                    decoder.dateDecodingStrategy = JSONDecoder.DateDecodingStrategy.supportMultipleDateFormats
+                    decoder.keyDecodingStrategy = .useDefaultKeys
```

Comment:
Have you considered making the decoder an instance or static property? Just to save having to init and configure it on every response?

Reply:
Good point! Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/21952/commits/d5ad31a40bc70afe69b87d4effe1082ca4a44c8f.